src/Init.ts

```diff
@@ -0,0 +1,88 @@
+import { execSync } from 'child_process';
+import { Errors } from './Types';
+
+import colors from 'colors'
+import fs from 'fs';
+import path from 'path';
+import yargs from 'yargs';
+
+
+export function checkRequirements() {
+    try {
+        return execSync('ffmpeg -version',
+        { stdio: ['pipe','pipe','ignore'] }).toString().split('\n')[0];
```

Comment:
this
@snobu

Reply:
This is our current implementation -

```typescript
export function checkRequirements() {
    try {
        const ffmpegVer = execSync('ffmpeg -version').toString().split('\n')[0];
        console.info(colors.green(`Using ${ffmpegVer}\n`));

    } catch (e) {
        process.exit(ERROR_CODE.MISSING_FFMPEG);
    }
}
```

Since stdout is captured in our variable and we don't really care about stderr since if we get it ffmpeg failed anyway, not sure if we need to manually pass the stdio array? @lukaarma Am i missing something?

Reply:
stderr is normally passed to destreamer stderr and is printed if ffpmeg is not installed, at least on windows and i thought that since we have our own error code we should use that and intercept any error that comes from ffmpeg in this case

another note, shouldn't we call process.exit() only from "main destreamer"? I was always told that it's a better practice to handle errors and exits up the chain than locally

Reply:
Yup, let's call process.exit() from destreamer.ts.

Reply:
mmh, but we have a catch, usually it is better to handle the catch or exit so that all the code is there and you don't have to look for it somewhere else